### PR TITLE
Dmm/update rubocop

### PR DESCRIPTION
**Pull request by Sgtpluck**
### Description of Changes:

- Updates .rubcop.yml to target current versions of ruby/rails
- Includes fixes to get `make lint` passing

### PR Checklist:

1. [ ] Have you linted and tested your code locally prior to submission?
2. [ ] Have you tagged the appropriate dev(s) for review?
3. [ ] Have you linked to any relevant tickets or conversations?

### PR Review Standards: 
- Consider using [Conventional Comments](https://conventionalcomments.org/) to ensure that your feedback is clear and actionable.
- Ideally, PRs should be reviewed by at least 2 team members.
- All PRs must be approved before being merged.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,6 +1,8 @@
 source 'https://rubygems.org'
 
-ruby "~> #{File.read(File.join(__dir__, '.ruby-version')).strip}"
+# If you update the rails version, please change the targeted
+# version in .rubocop.yml
```
Comment (Sgtpluck) [inline Gemfile:4]: this isn't ideal, since the ruby change happens in `.ruby-version`. there may be a better way to do this!

Comment (Sgtpluck) [reply Gemfile:4]: `Configuration files are pre-processed using the ERB templating mechanism` so i am gonna see if i can programmatically generate the version values

Comment (Sgtpluck) [reply Gemfile]: i did it! this concludes this conversation with myself. thanks for rubber ducking github.

File: app/models/tools/saml_request.rb [hunk 2]
```diff
@@ -46,12 +46,19 @@ def check_signature_validity
       @valid_signature ||= begin
 
         if certs.nil?
-          @errors.push('Could not find any certificates to use. Please add a certificate to your application configuration or paste one below.')
+          @errors.push(
+            'Could not find any certificates to use. Please add a ' +
+            'certificate to your application configuration or paste one below.',
+          )
```
Comment (zachmargolis) [inline app/models/tools/saml_request.rb:52]: what about heredoc syntax here?
```suggestion
          @errors.push(<<~EOS.squish)
            Could not find any certificates to use. Please add a
            certificate to your application configuration or paste one below.
          EOS
```

File: spec/controllers/tools_controller_spec.rb [hunk 3]
```diff
@@ -48,9 +48,9 @@
 
         it 'creates a flash warning' do
           msg = 'You have passed a logout request. Currently, this tool is for ' +
-          'Authentication requests only. Please try this ' +
-          '<a href="https://www.samltool.com/validate_logout_req.php" target="_blank">' +
-          'tool</a> to validate logout requests.'
+                'Authentication requests only. Please try this ' +
+                '<a href="https://www.samltool.com/validate_logout_req.php" target="_blank">' +
+                'tool</a> to validate logout requests.'
```
Comment (zachmargolis) [inline spec/controllers/tools_controller_spec.rb:53]: ditto for heredoc syntax?
```ruby
          msg = <<~EOS.squish
            You have passed a logout request. Currently, this tool is for
            Authentication requests only. Please try this
            <a href="https://www.samltool.com/validate_logout_req.php" target="_blank">
            tool</a> to validate logout requests
          EOS

File: spec/models/saml_request_spec.rb [hunk 4]
```diff
@@ -75,8 +75,10 @@
       end
 
       it 'adds an error' do
+        err = 'Could not find any certificates to use. Please add a ' +
+              'certificate to your application configuration or paste one below.'
```
Comment (zachmargolis) [inline spec/models/saml_request_spec.rb:79]: heredoc
```suggestion
        err = <<~EOS.squish
          Could not find any certificates to use. Please add a
          certificate to your application configuration or paste one below.
        EOS
```

**Review by zachmargolis — APPROVED** [review]

File: app/models/service_provider_certificate.rb [hunk 5]
```diff
@@ -7,9 +7,9 @@ def initialize(cert)
     @cert = cert
   end
 
-  def method_missing(name, *args, &block)
+  def method_missing(name, *, &)
     if cert.respond_to?(name)
-      cert.send(name, *args, &block)
+      cert.send(name, *, &)
```
[the remark this answers is not in the document]
Comment (Sgtpluck) [reply app/models/service_provider_certificate.rb:12]: i'll check which rule updated it, we can change it if it seems bad!

[the remark this answers is not in the document]
Comment (Sgtpluck) [reply app/models/service_provider_certificate.rb]: i disabled that rule in https://github.com/18F/identity-dashboard/pull/776/commits/8726d8eefc76b44db9fdc5eb9c75e8ae25dba821

File: app/models/tools/saml_request.rb (hunk 2 above)
Comment (Sgtpluck) [reply app/models/tools/saml_request.rb]: updated in https://github.com/18F/identity-dashboard/pull/776/commits/8726d8eefc76b44db9fdc5eb9c75e8ae25dba821

File: spec/controllers/tools_controller_spec.rb (hunk 3 above)
Comment (Sgtpluck) [reply spec/controllers/tools_controller_spec.rb]: updated in https://github.com/18F/identity-dashboard/pull/776/commits/8726d8eefc76b44db9fdc5eb9c75e8ae25dba821

File: spec/models/saml_request_spec.rb (hunk 4 above)
Comment (Sgtpluck) [reply spec/models/saml_request_spec.rb]: updated in https://github.com/18F/identity-dashboard/pull/776/commits/8726d8eefc76b44db9fdc5eb9c75e8ae25dba821

**Review by zachmargolis — APPROVED** [review]

**Pull request merged**